astropy/modeling/utils.py

```diff
@@ -22,6 +23,8 @@ class ExpressionTree:
     __slots__ = ['left', 'right', 'value', 'inputs', 'outputs']
 
     def __init__(self, value, left=None, right=None, inputs=None, outputs=None):
+        raise AstropyDeprecationWarning('The "ExpressionTree" class has been '
+                                        'deprecated and will be removed in v 4.1.')
```

Comment:
Deprecation in 4.0 followed by removal in 4.1 might be too soon. Maybe just state "in the future"?

Reply:
Agree, in general I don't think we need to say when it will be removed.